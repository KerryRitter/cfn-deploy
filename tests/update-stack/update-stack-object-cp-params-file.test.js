const AWS = require('aws-sdk-mock');
const lib = require('../..');


describe('cfn-deploy', () => {
  beforeAll(() => {
    AWS.mock('CloudFormation', 'validateTemplate', (params, callback) => {
      callback(null, {});
    });
    AWS.mock('CloudFormation', 'describeStacks', (params, callback) => {
      callback(null, {
        Stacks: [
          {
            StackStatus: 'UPDATE_COMPLETE',
          },
        ],
      });
    });
    AWS.mock('CloudFormation', 'createChangeSet', (params, callback) => {
      expect(params).toMatchSnapshot();
      callback(null, {
        Id: 'fake-changeset-id',
      });
    });
    AWS.mock('CloudFormation', 'describeChangeSet', (params, callback) => {
      callback(null, {
        ChangeSetId: 'fake-changeset-id',
      });
    });
    AWS.mock('CloudFormation', 'executeChangeSet', (params, callback) => {
      callback();
    });
    AWS.mock('CloudFormation', 'deleteChangeSet', (params, callback) => {
      callback();
    });
    AWS.mock('CloudFormation', 'waitFor', (event, params, callback) => {
      if (event === 'changeSetCreateComplete') {
        callback(null, {
          ChangeSetId: 'fake-changeset-id',
        });
      } else if (event === 'stackUpdateComplete') {
        callback(null, {
          Stacks: [
            {
              StackStatus: 'UPDATE_COMPLETE',
            },
          ],
        });
      }
    });
  });

  it('should successfully update existing stack', (done) => {
    const events = lib({
      region: 'us-east-1',
      stackName: 'existing-stack',
      template: './tests/templates/params-template.yaml',
      parameters: './tests/params/object-cp-params.json',
    });
    events.on('ERROR', err => done(err));
    events.on('COMPLETE', (data) => {
      expect(data).toMatchSnapshot();
      done();
    });
  });

  afterAll(() => {
    AWS.restore('CloudFormation');
  });
});
