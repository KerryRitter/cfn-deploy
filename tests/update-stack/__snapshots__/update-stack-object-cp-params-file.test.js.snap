// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cfn-deploy should successfully update existing stack 1`] = `
Object {
  "Capabilities": undefined,
  "ChangeSetName": "cfn-deploy",
  "ChangeSetType": "UPDATE",
  "Description": "Created with cfn-deploy",
  "Parameters": Array [
    Object {
      "ParameterKey": "S3BucketName",
      "ParameterValue": "from-object-cp-params-file",
    },
    Object {
      "ParameterKey": "DummyParam",
      "ParameterValue": "dummy-param",
    },
    Object {
      "ParameterKey": "OverwritableParam",
      "ParameterValue": "from-object-cp-params-file",
    },
    Object {
      "ParameterKey": "ParamSource",
      "ParameterValue": "object-cp-params.json",
    },
  ],
  "StackName": "existing-stack",
  "TemplateBody": "Parameters:
  S3BucketName:
    Description: \\"S3 Bucket name\\"
    Type: String
  DummyParam:
    Description: \\"Dummy parameter\\"
    Type: String

Resources:
  S3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: !Sub \\"\${AWS::StackName}-\${S3BucketName}\\"
",
}
`;

exports[`cfn-deploy should successfully update existing stack 2`] = `
Object {
  "type": "UPDATE",
}
`;
